jetty-server/src/main/java/org/eclipse/jetty/server/ResourceService.java

```diff
@@ -285,6 +289,35 @@ public void doGet(HttpServletRequest request, HttpServletResponse response)
         }
     }
 
+    private List<String> getPreferredEncodingOrder(HttpServletRequest request)
+    {
+        Enumeration<String> e = request.getHeaders(HttpHeader.ACCEPT_ENCODING.asString());
+        if (e==null || !e.hasMoreElements())
+            return emptyList();
+
+        String key=e.nextElement();
+        while (e.hasMoreElements())
+            key+='\n'+e.nextElement();
+
+        List<String> values=_preferredEncodingOrderCache.get(key);
+        if (values==null)
+        {
+            QuotedEncodingQualityCSV encodingQualityCSV = new QuotedEncodingQualityCSV(_precompressedFormats);
+            Enumeration<String> e2=request.getHeaders(HttpHeader.ACCEPT_ENCODING.asString());
+            while (e2.hasMoreElements())
+                encodingQualityCSV.addValue(e2.nextElement());
+            values=encodingQualityCSV.getValues();
+
+            // keep cache size in check even if we get strange/malicious input - with 4k headers this is still under 1 megabyte
+            if (_preferredEncodingOrderCache.size()>200)
```

Comment:
I see no reason why somebody would want to disable the cache. Or tune it. I think just selecting a good value is the best option. The balance is here between the number of unique accept-encoding header variants used in the real world (my guess is that that is < 30) and not having a memory leak exploitable for malicious entities.
If the cache overflows and resets (which is the current strategy), then we just spend a bit more CPU doing the next few requests parsing the headers but no difference in functionality is seen outside of this function,